npm/ipsm/ipsm.go

```diff
@@ -613,57 +605,59 @@ func (ipsMgr *IpsetManager) DestroyNpmIpsets() error {
 		return nil
 	}
 
-	entry := &ipsEntry{
-		operationFlag: util.IpsetFlushFlag,
-	}
-
+	destroyFailureCount := 0
```

Comment:
This part is a little tricky.
This `DestroyNpmIpsets` is called when NPM is first started or restarted due to some failures.
In restart, it has data plane information and if resetDataplane is failed, it will have stale information and mess up metric information. Then, NPM starts reinstalling dataplane with pod, ns, and network policy information. I am not sure this function correctly catch those possibilities. 
Would you please check it?

Reply:
So what happens exactly if resetDataplane fails with stale info? 
I thought we should keep counts for azure-npm ipsets that exist and fail to delete. We can set numIPSets to 0 all the time if we know that every existing azure-npm ipset will be deleted.

Reply:
Resolving from conversation in Teams